core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1278,12 +1278,8 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    */
   def broadcast[T: ClassTag](value: T): Broadcast[T] = {
     assertNotStopped()
-    if (classOf[RDD[_]].isAssignableFrom(classTag[T].runtimeClass)) {
-      // This is a warning instead of an exception in order to avoid breaking user programs that
-      // might have created RDD broadcast variables but not used them:
-      logWarning("Can not directly broadcast RDDs; instead, call collect() and "
-        + "broadcast the result (see SPARK-5063)")
-    }
+    require(!classOf[RDD[_]].isAssignableFrom(classTag[T].runtimeClass), "Can not directly " +
```

Comment:
Nit: rather than break this string just ...

```
require(!classOf[RDD[_]].isAssignableFrom(classTag[T].runtimeClass), 
  "Can not directly broadcast RDDs; instead, call collect() and broadcast the result.")
```